float64 x_tf1
float64 x_tf2
float64 x_tf3

float64 y_tf1
float64 y_tf2
float64 y_tf3

float64 phi_tf1

float64 phi_tf2

float64 phi_tf3
